lib/array.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+
+  def new_inject (a=self.shift, &block)
```

Comment:
Why have you called this parameter `a`?  it's very cryptic - try and use descriptive names at all times.  In the ruby version, this parameter is called `initial`